ophidian/src/ophidian/definedvenv.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python
+
+# coding=utf-8
+#
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis-python
+#
+# Most of this work is copyright (C) 2013-2017 David R. MacIver
+# (david@drmaciver.com), but it contains contributions by others. See
+# CONTRIBUTING.rst for a full list of people who may hold copyright, and
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at http://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+from __future__ import division, print_function, absolute_import
+
+import os
+import sys
+import shutil
+import hashlib
+import binascii
+import subprocess
+
+
+def fail(*args):
+    print(*args, file=sys.stderr)
+    sys.exit(1)
+
+
+PY3 = sys.version_info[0]
+
+
+def sha1(s):
+    if PY3:
+        s = s.encode('utf-8')
+    return hashlib.sha1(s).hexdigest()
+
+
+def call_or_die(*args, **kwargs):
+    kwargs['stdout'] = sys.stderr
+    result = subprocess.call(*args, **kwargs)
+    if result != 0:
+        sys.exit(result)
+
+
+def randhex():
+    result = binascii.hexlify(os.urandom(16))
+    if PY3:
+        result = result.decode('ascii')
+    return result
+
+
+if __name__ == '__main__':
+
+    if len(sys.argv) != 3:
+        fail('Usage: defined-env.py requirements target-dir')
+
+    _, requirements, target = sys.argv
+
+    if not os.path.exists(requirements):
+        fail('No such file', requirements)
```

Comment:
Slightly prefer a `try: ... except OSError: ...` around the attempt to read the file, to avoid window conditions with the file suddenly disappearing.